R/fit_CWCurve.R

```diff
@@ -722,22 +714,20 @@ fit_CWCurve<- function(
     col <- get("col", pos = .LuminescenceEnv)
 
     ##set plot frame
+    par(cex = cex.global)
     if(!inherits(fit, "try-error")){
       layout(matrix(c(1,2,3),3,1,byrow=TRUE),c(1.6,1,1), c(1,0.3,0.4),TRUE)
-      par(oma=c(1,1,1,1),mar=c(0,4,3,0),cex=cex.global)
-    }else{
-      par(cex=cex.global)
+      par(oma = c(1, 1, 1, 1), mar = c(0, 4, 3, 0))
     }
 
-
     ##==uppper plot==##
     ##open plot area
 
     plot(NA,NA,
          xlim=c(min(x),max(x)),
-         ylim=if(log=="xy"){c(1,max(y))}else{c(0,max(y))},
-         xlab=if(!inherits(fit, "try-error")){""}else{xlab},
-         xaxt=if(!inherits(fit, "try-error")){"n"}else{"s"},
+         ylim = c(ifelse(log == "xy", 1, 0), max(y)),
+         xlab = ifelse(inherits(fit, "try-error"), xlab, ""),
+         xaxt = ifelse(inherits(fit, "try-error"), "s", "n"),
```

Comment:
I had avoided ifelse() because it evaluates differently from if () {} else {}. In particular there is a recycling going on, please see the documentation and the statement is slower.

Reply:
It's problematic for vectors and such, but for scalars no recycling is possible.

Reply:
Agreed, however, two reasons: 

- if we expect a scalar anyway, there is no need for `ifelse()`
- the call is an order of magnitude slower and I don't see any advantage using it here except that it looks more concise

```r
f_ifelse <- function() ifelse(TRUE, "", "")
f_standard <- function() if(TRUE) "" else ""
microbenchmark::microbenchmark(f_ifelse(), f_standard())
```

```
Unit: nanoseconds
         expr min  lq    mean median  uq    max neval cld
   f_ifelse() 615 656 4477.20    656 697 376913   100   a
 f_standard()  41  41 4363.22     82  82 428696   100   a
```